### PR TITLE
Implement ListOrganizations

### PR DESCRIPTION
**Pull request by gliptak**
https://github.com/cloudflare/cloudflare-go/issues/35

@jamesog @elithrar Please review and guide on pagination.

One thought would be to rename ```ResultInfo``` to ```Paginator``` and add a second method signature taking in ```Paginator```. Or variadic args as per https://github.com/cloudflare/cloudflare-go/pull/4 ?

Thanks

### CODE REVIEW

**Comment by gliptak** [comment]
Could you point out the preferred approach? Thanks

**Comment by jamesog** [comment]
Hi @gliptak - thanks for this!

Pagination we've discussed a bit in #6 and #78. It's going to be a breaking change to implement it so it's probably worth doing everything at once. I've got a WIP I'm testing at the moment but needs a bit more work before I submit it.

For now I think this PR is OK. I recently updated the `DNSRecords()` method (in #102) to do the pagination itself and return all records, until we've implemented pagination properly.

File: organizations.go [hunk 1]
```diff
@@ -9,9 +15,28 @@ type Organization struct {
 	Roles       []string `json:"roles,omitempty"`
 }
 
-// OrganizationResponse represents the response from the Organization endpoint.
-type OrganizationResponse struct {
+// organizationResponse represents the response from the Organization endpoint.
+type organizationResponse struct {
 	Response
 	Result     []Organization `json:"result"`
 	ResultInfo ResultInfo     `json:"result_info"`
```
Comment (jamesog) [inline organizations.go]: Go lets you omit the second `ResultInfo` here but I'm OK either way having it or not.

Comment (gliptak) [inline organizations.go]: Removed ResultInfo

**Pull request merged**